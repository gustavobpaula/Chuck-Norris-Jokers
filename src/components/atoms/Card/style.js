import styled from 'styled-components';
import designSystem from '../../../designSystem';

export const Card = styled.div`
  background-color: ${designSystem.get('colors.color3')};
  border-radius: 15px;
  cursor: pointer;
  padding: 20px;
  text-align: center;
  transition: background-color 0.3s ease, transform 0.3s ease;
  z-index: 1;

  &:hover {
    background-color: ${designSystem.get('colors.color2')};
    transform: scale(1.05);
    z-index: 2;
  }
`;
