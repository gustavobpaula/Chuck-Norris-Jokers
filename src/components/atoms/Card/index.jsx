import React from 'react';
import PropTypes from 'prop-types';
import * as Styled from './style';

const Card = ({ children }) => <Styled.Card>{children}</Styled.Card>;

Card.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Card;
