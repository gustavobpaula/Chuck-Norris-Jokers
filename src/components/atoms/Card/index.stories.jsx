import React from 'react';
import Card from './index';

export default {
  component: Card,
  title: 'Atoms/Card',
};

export const element = () => <Card>Card Component</Card>;
element.story = {
  name: 'Default',
};
