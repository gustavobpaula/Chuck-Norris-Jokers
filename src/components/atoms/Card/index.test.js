import React from 'react';
import { mount } from 'enzyme';

import Card from './index';

describe('<Card />', () => {
  it('should mount properly', async () => {
    const wrapper = mount(<Card>Card Component</Card>);

    expect(wrapper).toMatchSnapshot();
  });
});
