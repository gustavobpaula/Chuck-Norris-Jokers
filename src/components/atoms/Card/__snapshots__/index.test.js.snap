// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should mount properly 1`] = `
<Card>
  <style__Card>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "style__Card-sc-7a5vgg-0",
            "isStatic": false,
            "lastClassName": "hEuRRR",
            "rules": Array [
              "background-color:",
              "#145374",
              ";border-radius:15px;cursor:pointer;padding:20px;text-align:center;transition:background-color 0.3s ease,transform 0.3s ease;z-index:1;&:hover{background-color:",
              "#5588a3",
              ";transform:scale(1.05);z-index:2;}",
            ],
          },
          "displayName": "style__Card",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "style__Card-sc-7a5vgg-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="style__Card-sc-7a5vgg-0 hEuRRR"
      >
        Card Component
      </div>
    </StyledComponent>
  </style__Card>
</Card>
`;
